lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -121,6 +122,42 @@ export class ResponseHandler {
         return ResponseHandler.generateAuthenticationResult(cacheRecord, idTokenObj, false, requestStateObj);
     }
 
+    handleBrokeredServerTokenResponse(serverTokenResponse: ServerAuthorizationTokenResponse, authority: Authority, cachedNonce?: string, cachedState?: string): BrokerAuthenticationResult {
+        // create an idToken object (not entity)
+        const idTokenObj = new IdToken(serverTokenResponse.id_token, this.cryptoObj);
+
+        // token nonce check (TODO: Add a warning if no nonce is given?)
+        if (!StringUtils.isEmpty(cachedNonce)) {
+            if (idTokenObj.claims.nonce !== cachedNonce) {
+                throw ClientAuthError.createNonceMismatchError();
+            }
+        }
+
+        // save the response tokens
+        let requestStateObj: RequestStateObject = null;
+        if (!StringUtils.isEmpty(cachedState)) {
+            requestStateObj = ProtocolUtils.parseRequestState(this.cryptoObj, cachedState); 
+        }
+
+        // Get creds to send to child
+        const cacheRecord = this.generateCacheRecord(serverTokenResponse, idTokenObj, authority, requestStateObj && requestStateObj.libraryState);
+        const refreshTokenRecord: CacheRecord = {
+            accessToken: null,
+            idToken: null,
+            refreshToken: cacheRecord.refreshToken,
+            account: null
+        };
+
+        // Save refresh token
+        this.cacheStorage.saveCacheRecord(refreshTokenRecord);
```

Comment:
Can't you just use `this.cacheStorage.saveCredential(cacheRecord.refreshToken)` instead?

Reply:
Good point will change.